src/Dialog/Dialog.js

```diff
@@ -1,14 +1,15 @@
 // @flow weak
 
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { PureComponent, Element } from 'react';
```

Comment:
I don't think that we need any pure logic inside Material-UI, our components are too low level to be efficient.